src/core/mac/mac_frame.cpp

```diff
@@ -1048,6 +1070,140 @@ void TxFrame::ProcessTransmitAesCcm(const ExtAddress &aExtAddress)
 #endif // OPENTHREAD_RADIO
 }
 
+otError TxFrame::GenerateImmAck(const RxFrame &aFrame, bool aIsFramePending)
+{
+    uint16_t fcf = 0;
```

Comment:
How about initializing the `fcf` with all the always-used fields (to remove the `fcf |= kFcfFrameAck`, etc)?
```suggestion
    uint16_t fcf = kFcfFrameAck | aFrame.GetVersion();
```